test/api/handlers_test.go

```diff
@@ -302,6 +304,70 @@ func testUserPUT(t *testing.T, at *test.AccountsTester) {
 	}
 }
 
+// testUserDELETE tests the DELETE /user endpoint.
+func testUserDELETE(t *testing.T, at *test.AccountsTester) {
+	u, c, err := test.CreateUserAndLogin(at, t.Name())
+	if err != nil {
+		t.Fatal("Failed to create a user and log in:", err)
+	}
+	// Delete the user.
+	at.Cookie = c
+	defer func() { at.Cookie = nil }()
+	r, _, err := at.Delete("/user", nil)
+	if err != nil || r.StatusCode != http.StatusNoContent {
```

Comment:
I don't think we should be getting `StatusNoContent`. If a user exists, but doesn't have content, they should just be deleted regardless of if there is content of not. 

Should add verification that the user is in fact gone.

Reply:
1. Status No Content means that the operation was performed successfully and there is no data that's supposed to be returned. This is what our `WriteSuccess` method does both in accounts and in skyd.
2. Explicit verification that the user is gone would be redundant, as we try to create a user with the exact same data 3 lines later.